.pipelines/scripts/run-e2e.sh

```diff
@@ -62,26 +53,31 @@ elif [[ "${CLUSTER_TYPE}" == "autoscaling-multipool" ]]; then
   export AZURE_LOADBALANCER_SKU=standard
 fi
 
-CLUSTER_CONFIG_KEYS=("AKS_CLUSTER_ID" "CLUSTER_NAME" "AZURE_LOCATION" "KUBERNETES_VERSION" "AZURE_CLIENT_ID" "AZURE_CLIENT_SECRET" "CUSTOM_CONFIG")
-CLUSTER_CONFIG_VALS=("${AKS_CLUSTER_ID}" "${CLUSTER_NAME}" "${AZURE_LOCATION}" "${KUBERNETES_VERSION}" "${AZURE_CLIENT_ID}" "${AZURE_CLIENT_SECRET}" "${CUSTOM_CONFIG}")
-for i in "${!CLUSTER_CONFIG_KEYS[@]}";do
-  sed -i "s|${CLUSTER_CONFIG_KEYS[$i]}|${CLUSTER_CONFIG_VALS[$i]}|g" "${CLUSTER_CONFIG_PATH}"
-done
+pushd kubetest2-aks
+go get -d sigs.k8s.io/kubetest2@latest
+go install sigs.k8s.io/kubetest2@latest
+make deployer
+sudo GOPATH="/home/vsts/go" make install
+popd
+kubetest2 aks --up --rgName "${RESOURCE_GROUP}" \
+--location "${AZURE_LOCATION}" \
+--config "${CLUSTER_CONFIG_PATH}" \
+--customConfig "${REPO_ROOT}/.pipelines/templates/customconfiguration.json" \
+--clusterName "${CLUSTER_NAME}" \
+--ccmImageTag "${IMAGE_TAG}" \
+--k8sVersion "${KUBERNETES_VERSION}"
 
-echo "Creating an AKS cluster in resource group ${RESOURCE_GROUP}"
-CREATION_DATE="$(date +%s)"
-az group create --name "${RESOURCE_GROUP}" -l "${AZURE_LOCATION}" --tags "creation_date=${CREATION_DATE}" "usage=aks-cluster-e2e"
-curl -i -X PUT -k -H "Authorization: Bearer ${TOKEN}" \
-    -H "Content-Type: application/json" \
-    -H "AKSHTTPCustomFeatures: Microsoft.ContainerService/EnableCloudControllerManager" \
-    "https://management.azure.com${AKS_CLUSTER_ID}?api-version=2022-04-02-preview" \
-    -d "$(cat ${CLUSTER_CONFIG_PATH})"
+export KUBECONFIG="${REPO_ROOT}/_kubeconfig/${RESOURCE_GROUP}_${CLUSTER_NAME}.kubeconfig"
+if [[ ! -f "${KUBECONFIG}" ]]; then
+  echo "kubeconfig not exists"
+  exit 1
+fi
+
+for i in `seq 1 6`; do
```

Comment:
could you add a comment why get pods for a few times here?

Reply:
Updated.